### DISCUSSION TITLE
IDEA 各项代码都是比较固定的，能否智能化接入IDEA各个组件功能

### DISCUSSION THREAD

**Discussion by c913168497** [unit-mesh/auto-dev#185] [Ideas]
比如我要crud, 我可以自己流程编排
1. 正在寻找数据库连接配置文件 （搜索全局文件）
2. 正在分析数据库连接配置 （自定义分析语义）
3. 提示人工确认 （人工交互编辑）
4. 正在选择匹配的表（数据库操作（搜索表名））
5. 正在分析当前文件目录结构（全局文件夹路径分析）
6. 提示人工确认 controller, service,mapper 层路径（人工交互编辑）
5. 正在生成对应文件（文件内容生成输出）
7. 生成结束（人工交互编辑）

**Comment by c913168497** [comment]
我思考了一下，我想你的 devin 大概最终其实想实现的效果应该就是类似这种，半自动化AI 辅助驾驶，类似可编辑的工作流

**Reply by phodal** [reply]
是的  DevIns  就是针对这个功能设计的。

**Reply by c913168497** [reply]
我理解了你的代码，但是我觉得存在一定的不合理，我尝试修改 在同一个 devin 文件中 可以执行多个操作，用于完成对于devin的设想


但是我觉得仍然没有达到我心中的想法，我觉得应该需要从编译器这块开始改造 psi 这一部分，我相信 ai 的接口调用费用会越来越便宜，我们这种副驾驶开发，结合 idea 是用很大潜力的，大佬有时间优化一下 这块的逻辑吗？

**Reply by phodal** [reply]
我短期有别的优先级，你可以设计一下，然后来来个完整的方案 => 创建个新的 issue
